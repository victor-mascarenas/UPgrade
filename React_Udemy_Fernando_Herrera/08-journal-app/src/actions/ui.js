import { types } from "../reducers/types/types";

export const setError = (msg) => {
    return ({
        type: types.UI_SET_ERROR,
        payload: {
            errorMsg: msg
        }
    });
};

export const removeError = () => {
    return ({
        type: types.UI_REMOVE_ERROR
    });
};