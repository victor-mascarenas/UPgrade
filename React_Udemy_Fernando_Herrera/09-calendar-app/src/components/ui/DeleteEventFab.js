import React from 'react'
import { useDispatch } from 'react-redux';
import { eventStartDelete } from '../../actions/events';

export const DeleteEventFab = () => {
    const dispatch = useDispatch();

    const onClick = () => {
        dispatch(eventStartDelete());
    };

    return (
        <button className='btn btn-danger fab-danger' onClick={onClick}>
            <i className='fas fa-trash'></i>
            <span> Borrar</span>
        </button>
    )
}
