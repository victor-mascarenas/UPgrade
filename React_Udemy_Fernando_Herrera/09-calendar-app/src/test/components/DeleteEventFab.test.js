import React from 'react';
import {mount} from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { DeleteEventFab } from '../../components/ui/DeleteEventFab';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
const store = mockStore(initState);
const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab/>
    </Provider>
);

store.dispatch = jest.fn();

describe('Pruebas en DeleteEventFab component', () => {
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
});