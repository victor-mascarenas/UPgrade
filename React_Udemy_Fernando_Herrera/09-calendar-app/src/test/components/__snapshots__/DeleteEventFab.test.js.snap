// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DeleteEventFab component Debe mostrarse correctamente 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar
  </span>
</button>
`;
